aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1114,10 +1113,14 @@ Tensor &any_out(const Tensor &self, int64_t dim, bool keepdim, Tensor &result) {
 }
 
 Tensor &amin_out(const Tensor& self, IntArrayRef dim, bool keepdim, Tensor& result) {
-  TORCH_CHECK(self.scalar_type() == result.scalar_type(), "Illegal dtype for self, and out:", self.scalar_type(), result.scalar_type());
+  TORCH_CHECK(self.scalar_type() == result.scalar_type(), "Illegal dtype for self, and out:",
+    self.scalar_type(), result.scalar_type());
+  zero_numel_check_dims(self, dim, "amin()");
+
   auto iter = make_reduction("amin", result, self, dim, keepdim, self.scalar_type());
```

Comment:
not for this PR, but it would nice to rename `make_reduction` to make it clear it also resizes outputs

Reply:
OK I'll open an issue for that and link it to this comment.

Reply:
https://github.com/pytorch/pytorch/issues/56764